src/mongodb.appup.src

```diff
@@ -1,184 +1,95 @@
     {load_module, mc_worker_logic, brutal_purge, soft_purge, []},
+    {load_module, mongo_protocol, brutal_purge, soft_purge, []},
+    {load_module, mc_super_sup, brutal_purge, soft_purge, []},
+    {load_module, mc_worker_pid_info, brutal_purge, soft_purge, []},
     {load_module, mc_monitor, brutal_purge, soft_purge, []},
     {load_module, mc_server, brutal_purge, soft_purge, []},
-    {load_module, mc_topology, brutal_purge, soft_purge, []},
-    {load_module, mc_topology_logics, brutal_purge, soft_purge, []}
-  ]},
-  {<<"3.0.[0-7]">>, [
-    {restart_application, mongodb}
-  ]},
-  {<<".*">>, []}
- ],
- [{"3.0.16", [
-    {load_module, mongoc, brutal_purge, soft_purge, []},
-    {load_module, mc_worker_api, brutal_purge, soft_purge, []},
-    {load_module, mongo_api, brutal_purge, soft_purge, []}
-  ]},
-  {"3.0.15", [
-    {load_module, mongoc, brutal_purge, soft_purge, []},
-    {load_module, mc_worker_api, brutal_purge, soft_purge, []},
-    {load_module, mongo_api, brutal_purge, soft_purge, []},
-    {load_module, mc_topology_logics, brutal_purge, soft_purge, []}
-  ]},
-  {"3.0.14", [
-    {load_module, mc_worker_api, brutal_purge, soft_purge, []},
-    {load_module, mongo_api, brutal_purge, soft_purge, []},
+    {load_module, mc_utils, brutal_purge, soft_purge, []},
     {load_module, mc_topology_logics, brutal_purge, soft_purge, []},
-    {load_module, mongoc, brutal_purge, soft_purge, []}
-  ]},
-  {"3.0.13", [
-    {load_module, mc_worker_api, brutal_purge, soft_purge, []},
-    {load_module, mongo_api, brutal_purge, soft_purge, []},
-    {load_module, mongoc, brutal_purge, soft_purge, []},
-    {load_module, mc_auth_logic, brutal_purge, soft_purge, []},
+    {load_module, mc_topology, brutal_purge, soft_purge, []},
     {load_module, mc_worker, brutal_purge, soft_purge, []},
-    {load_module, mc_topology_logics, brutal_purge, soft_purge, []},
-    {load_module, mc_util, brutal_purge, soft_purge, []}
-  ]},
-  {"3.0.12", [
     {load_module, mc_worker_api, brutal_purge, soft_purge, []},
     {load_module, mongo_api, brutal_purge, soft_purge, []},
-    {load_module, mongoc, brutal_purge, soft_purge, []},
-    {load_module, mc_auth_logic, brutal_purge, soft_purge, []},
-    {load_module, mc_worker, brutal_purge, soft_purge, []},
-    {load_module, mc_util, brutal_purge, soft_purge, []},
-    {load_module, mc_topology_logics, brutal_purge, soft_purge, []},
-    {load_module, mc_topology, brutal_purge, soft_purge, []}
+    {load_module, mongoc, brutal_purge, soft_purge, []}
   ]},
-  {"3.0.11", [
-    {delete_module, mc_util},
-    {load_module, mc_worker_api, brutal_purge, soft_purge, []},
-    {load_module, mongo_api, brutal_purge, soft_purge, []},
-    {load_module, mongoc, brutal_purge, soft_purge, []},
+  {<<".*">>, []}
+ ],
+ [
+   {<<"3.0.[0-7]">>, [
+    {restart_application, mongodb}
+   ]},
+   {<<"3.0.([89]|10)">>, [
+    {remove_module, mc_util},
+    {remove_module, mc_worker_pid_info},
```

Comment:
Does the `remove_module` instruction exist?  I couldn't spot it here: https://www.erlang.org/doc/man/appup.html

Perhaps:

```suggestion
    {delete_module, mc_util},
    {delete_module, mc_worker_pid_info},
```

?

Also, these removals probably should go at the end of the instructions, since it's possible that other modules yet to be reloaded are using them (maybe even processes running, like worker pid info).

Reply:
Thanks, my bad, will fix in new commit.

Reply:
Fixed both issues in [39060dc](https://github.com/emqx/mongodb-erlang/commit/39060dc39627daf9d666e3f4556fadce35bbe205)